plugin/pulpcore/plugin/download/futures/batch.py

```diff
@@ -89,29 +96,13 @@ def download(self):
                 The iterator will yield the download `Plan` in the order completed.
         """
         log.debug(_('%(batch)s - download started'), {'batch': self})
-        self.feeder.start()
+        for i, download in enumerate(self.downloads):
```

Comment:
If getting downloads out of this iterator is even slightly expensive (as described in another comment), it's possible that the ``completed()`` method above might get called before this iteration is finished. So I think the lock should be acquired here as well.

Reply:
Agreed.  Done.